sync/handler.go

```
@@ -64,29 +64,65 @@ func newMiniBlockRequestHandler(msh *mesh.Mesh, logger log.Log) func(msg []byte)
 			return nil
 		}
 
-		logger.Debug("return block %v", blk)
+		logger.Debug("return block %v", blk.ID())
 
 		return bbytes
 	}
 }
 
 func newTxsRequestHandler(msh *mesh.Mesh, logger log.Log) func(msg []byte) []byte {
 	return func(msg []byte) []byte {
-		var txid types.TransactionId
-		copy(txid[:], msg[:32])
-		logger.Info("handle tx request %v", hex.EncodeToString(txid[:]))
-		txs, _ := msh.GetTransactions([]types.TransactionId{txid})
+		var txids []types.TransactionId
+		err := types.BytesToInterface(msg, &txids)
+		if err != nil {
+			logger.Error("Error marshalling request", err)
+			return nil
+		}
+		logger.Info("handle tx request ")
+		txs, _ := msh.GetTransactions(txids)
```

error is ignored here, please add it to log output